index.js

```diff
@@ -233,12 +233,14 @@ module.exports = {
       let config = type === 'head' ? this._config : this._configForTest;
       let policyString = type === 'head' ? this._policyString : this._policyStringForTest;
 
-      this.ui.writeWarnLine(
-        'Content Security Policy does not support report only mode if delivered via meta element. ' +
-        "Either set `ENV['ember-cli-content-security-policy'].reportOnly` to `false` or remove `'meta'` " +
-        "from `ENV['ember-cli-content-security-policy'].delivery`.",
-        config.reportOnly
-      );
+      if (this._config.reportOnly && this._config.delivery.indexOf('meta') !== -1) {
+        this.ui.writeWarnLine(
+          'Content Security Policy does not support report only mode if delivered via meta element. ' +
+          "Either set `ENV['ember-cli-content-security-policy'].reportOnly` to `false` or remove `'meta'` " +
+          "from `ENV['ember-cli-content-security-policy'].delivery`.",
+          config.reportOnly
+        );
+      }
```

Comment:
Ups. I fear I used the `writeWarnLine` of [`console-ui`](https://github.com/ember-cli/console-ui) as if it is `warn` from `@ember/debug`. :grimacing: `writeWarnLine` doesn't seem to have a second argument, that controls if the warning is shown or not.

Also the message is outdated. It still references the deprecated configuration in `config/environment.js`. It should refer to `config/content-security-policy.js`. Seems like I missed to update it.

Do you have time to fix the message and remove the second argument?